.github/workflows/tests.yml

```
@@ -0,0 +1,152 @@
+          - python: 3.5
+            geos: 3.5.2
+            numpy: 1.10.4
+            speedups: 1
+          - python: 3.5
+            geos: 3.5.2
+            numpy: 1.10.4
+            speedups: 0
+          - python: 3.5
+            geos: 3.5.2
+            speedups: 0
+          # 2017
+          - python: 3.6
+            geos: 3.6.4
+            numpy: 1.13.3
+            speeedups: 1
+          - python: 3.6
+            geos: 3.6.4
+            numpy: 1.13.3
+            speeedups: 0
+          - python: 3.6
+            geos: 3.6.4
+            speeedups: 0
+          # 2018
+          - python: 3.7
+            geos: 3.7.3
+            numpy: 1.15.4
+            speedups: 1
+          - python: 3.7
+            geos: 3.7.3
+            numpy: 1.15.4
+            speedups: 0
+          - python: 3.7
+            geos: 3.7.3
+            speedups: 0
+          # 2019
+          - python: 3.8
+            geos: 3.8.1
+            numpy: 1.17.5
+            speedups: 1
+          - python: 3.8
+            geos: 3.8.1
+            numpy: 1.17.5
+            speedups: 0
+          - python: 3.8
+            geos: 3.8.1
+            speedups: 0
+          # 2020
+          - python: 3.9
+            geos: 3.9.0
+            numpy: 1.19.5
+            speedups: 1
+          - python: 3.9
+            geos: 3.9.0
+            numpy: 1.19.5
+            speedups: 0
+          - python: 3.9
+            geos: 3.9.0
+            speedups: 0
+          # dev
+          - python: 3.9
+            geos: master
+            speedups: 1
+
+    env:
+      GEOS_VERSION: ${{ matrix.geos }}
+      GEOS_INSTALL: ${{ github.workspace }}/geosinstall/geos-${{ matrix.geos }}
+      SPEEDUPS: ${{ matrix.speedups }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Cache GEOS and pip packages
+        uses: actions/cache@v2
+        with:
+          key: ${{ runner.os }}-geos-${{ matrix.geos }}
+          path: |
+            ~/.cache/pip
+            ${{ github.workspace }}/geosinstall
+
+      - name: Install GEOS
+        shell: bash
+        run: |
+          ./ci/install_geos.sh
+          cd ${{ github.workspace }}
+
+      - name: Install python dependencies
+        shell: bash
+        run: |
+          pip install --disable-pip-version-check --upgrade pip
+          pip install --upgrade wheel
+          if [ "$GEOS_VERSION" = "master" ]; then
+            pip install --upgrade --pre Cython numpy;
+          elif [ "$SPEEDUPS" == "1" ]; then
+            pip install --install-option="--no-cython-compile" cython;
+            pip install --upgrade numpy==${{ matrix.numpy }};
+          elif [ -n "${{ matrix.numpy }}" ]; then
+            pip install --upgrade numpy==${{ matrix.numpy }};
+          fi
+          pip install --upgrade coveralls pytest-cov pytest>=3.8
```

```suggestion
          pip install --upgrade coveralls pytest-cov pytest
```
(this was my mistake a while ago, which redirects some output instead of specifying a minimum version)